### PR TITLE
Accept answer crud

### PR DESCRIPTION
**Pull request by heinosasshallik**
Added functionality for accepting answers

### CODE REVIEW

File: api/src/main/java/com/qaengine/controllers/AnswerController.java [hunk 1]
```diff
@@ -90,4 +90,32 @@ protected Answer downvoteAnswer(
...
     }
+
+
+    @CrossOrigin()
+    @PutMapping("questions/{questionId}/acceptAnswer/{answerId}")
+    protected Answer acceptAnswer(
+            @PathVariable Long questionId,
+            @PathVariable Long answerId
+    ) {
+        Answer answer = answerService.getAnswer(answerId);
```
Comment (priitkaard) [inline api/src/main/java/com/qaengine/controllers/AnswerController.java:96]: Just use answerId, later we can verify with authentication if user is allowed to accept the answer or not. Each answer has unique id anyway, and it's connected to only one question.

File: api/src/main/java/com/qaengine/controllers/AnswerController.java [hunk 2]
```diff
@@ -90,4 +90,32 @@ protected Answer downvoteAnswer(
...
+        Question question = questionService.getQuestion(questionId);
+        if (!answer.getQuestionId().equals(questionId)) {
+            throw new BadRequestException("Answer does not belong to specified question.");
+        }
+        this.revertAnswerAccepted(question.getId());
```
Comment (priitkaard) [inline api/src/main/java/com/qaengine/controllers/AnswerController.java]: You are reusing controller logic again :) try to use service or repository

File: api/src/main/java/com/qaengine/controllers/AnswerController.java [hunk 3]
```diff
@@ -90,4 +90,31 @@ protected Answer downvoteAnswer(
...
+
+
+    @CrossOrigin()
+    @PutMapping("answers/{answerId}/accept")
+    protected Answer acceptAnswer(
+            @PathVariable Long answerId
+    ) {
+        Answer answer = answerService.getAnswer(answerId);
+        Question question = questionService.getQuestion(answer.getQuestionId());
+        if (!answer.getQuestionId().equals(answer.getQuestionId())) {
```
Comment (priitkaard) [inline api/src/main/java/com/qaengine/controllers/AnswerController.java]: What is that :D

**Review by priitkaard — APPROVED** [review]

File: api/src/main/java/com/qaengine/controllers/AnswerController.java (hunk 3 above)
Comment (heinosasshallik) [reply api/src/main/java/com/qaengine/controllers/AnswerController.java]: That is a blunder xD

**Pull request merged**